src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/NotificationOption2.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+#if CODE_STYLE
+using WorkspacesResources = Microsoft.CodeAnalysis.CodeStyleResources;
+#else
+using System.Security;
+using Roslyn.Utilities;
+#endif
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    /// <summary>
+    /// Offers different notification styles for enforcing
+    /// a code style. Under the hood, it simply maps to <see cref="DiagnosticSeverity"/>
+    /// </summary>
+    /// <remarks>
+    /// This also supports various properties for databinding.
+    /// </remarks>
+    /// <completionlist cref="NotificationOption2"/>
+    internal class NotificationOption2
+    {
+        public string Name { get; set; }
```

Comment:
```suggestion
        public string Name { get; }
```

Reply:
I have retained the API surface and behavior that was supported in the public API. We can consider the breaking/obsoleting changes in a follow-up PR.

Reply:
This setter is _invalid_ for this instance (invoking it will corrupt state of singleton objects). This is a new object which is not bound by previous API requirements, so we should not include these setters.

Reply:
Tracked removal with https://github.com/dotnet/roslyn/issues/42550#issue-584080273